first-party/jib-native-image-extension-gradle/src/test/java/com/google/cloud/tools/jib/gradle/extension/nativeimage/JibNativeImageExtensionTest.java

```diff
@@ -0,0 +1,237 @@
+    Map<String, String> properties = Collections.singletonMap("imageName", "theExecutable");
+    tempFolder.newFolder("native/nativeCompile");
+    tempFolder.newFile("native/nativeCompile/theExecutable");
+
+    ContainerBuildPlan buildPlan =
+        ContainerBuildPlan.builder()
+            .setEntrypoint(Collections.singletonList("to be overwritten"))
+            .build();
+    ContainerBuildPlan newPlan =
+        new JibNativeImageExtension()
+            .extendContainerBuildPlan(buildPlan, properties, Optional.empty(), gradleData, logger);
+
+    assertEquals(Collections.singletonList("/app/theExecutable"), newPlan.getEntrypoint());
+  }
+
+  @Test
+  public void testEntrypoint_setByJib() throws JibPluginExtensionException, IOException {
+    Map<String, String> properties = Collections.singletonMap("imageName", "theExecutable");
+    tempFolder.newFolder("native/nativeCompile/");
+    tempFolder.newFile("native/nativeCompile/theExecutable");
+
+    when(jibContainer.getEntrypoint()).thenReturn(Collections.singletonList("non-empty"));
+
+    ContainerBuildPlan buildPlan =
+        ContainerBuildPlan.builder().setEntrypoint(Collections.singletonList("set by Jib")).build();
+    ContainerBuildPlan newPlan =
+        new JibNativeImageExtension()
+            .extendContainerBuildPlan(buildPlan, properties, Optional.empty(), gradleData, logger);
+
+    assertEquals(Collections.singletonList("set by Jib"), newPlan.getEntrypoint());
+  }
+
+  @Test
+  public void testAppRoot() throws JibPluginExtensionException, IOException {
+    Map<String, String> properties = Collections.singletonMap("imageName", "theExecutable");
+    tempFolder.newFolder("native/nativeCompile/");
+    tempFolder.newFile("native/nativeCompile/theExecutable");
+
+    when(jibContainer.getAppRoot()).thenReturn("/new/root");
+
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+    ContainerBuildPlan newPlan =
+        new JibNativeImageExtension()
+            .extendContainerBuildPlan(buildPlan, properties, Optional.empty(), gradleData, logger);
+
+    assertEquals(Arrays.asList("/new/root/theExecutable"), newPlan.getEntrypoint());
+
+    assertEquals(1, newPlan.getLayers().size());
+    FileEntriesLayer layer = (FileEntriesLayer) newPlan.getLayers().get(0);
+    assertEquals(Arrays.asList("/new/root/theExecutable"), layerToExtractionPaths(layer));
+  }
+
+  @Test
+  public void testNoExecutableNameDetected() {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+
+    try {
+      new JibNativeImageExtension()
+          .extendContainerBuildPlan(
+              buildPlan, Collections.emptyMap(), Optional.empty(), gradleData, logger);
+      fail();
+    } catch (JibPluginExtensionException ex) {
+      assertEquals(
+          "cannot auto-detect native-image executable name; consider setting 'imageName' property",
+          ex.getMessage());
+    }
```

Comment:
How about we do this:

```suggestion
    Exception exception = assertThrows(JibPluginExtensionException.class, () -> new JibNativeImageExtension()
          .extendContainerBuildPlan(
              buildPlan, Collections.emptyMap(), Optional.empty(), gradleData, logger));
    assertThat(exception).hasMessageThat().isEqualTo( "cannot auto-detect native-image executable name; consider setting 'imageName' property");
```
And add the following import statement above: ` import static org.junit.Assert.assertThrows;`

Reply:
Intelij reckons that 

> org.junit.Assert.assertThrows

 is deprecated. Any thoughts?

Reply:
Maybe we want to add google truth to the class path and use:

> import static com.google.common.truth.Truth.assertThat;